Saucedemo/Core/Wrappers/UIElement.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.ObjectModel;
+using System.Drawing;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Interactions;
+
+namespace Saucedemo.Core.Wrappers
+{
+    public class UiElement : IWebElement
+    {
+        private readonly IWebElement _webElementImplementation;
+        private readonly IWebDriver _driver;
+        private By _by;
+        private readonly Actions _actions;
+        private readonly IJavaScriptExecutor _javaScriptExecutor;
+
+        public UiElement(IWebDriver webDriver, By by)
+        {
+            _driver = webDriver;
+            _actions = new Actions(webDriver);
+            _javaScriptExecutor = (IJavaScriptExecutor) _driver;
+            _by = by;
+            _webElementImplementation = webDriver.FindElement(by);
+        }
+
+        public UiElement(IWebDriver webDriver, IWebElement webElement)
+        {
+            _webElementImplementation = webElement;
+            _driver = webDriver;
+            _actions = new Actions(webDriver);
+            _webElementImplementation = webElement;
+        }
+
+        public void Hover()
+        {
+            _actions.MoveToElement(_webElementImplementation).Build().Perform();
+        }
+        
+        public IWebElement FindElement(By @by)
+        {
+            return _webElementImplementation.FindElement(@by);
+        }
+
+        public ReadOnlyCollection<IWebElement> FindElements(By @by)
+        {
+            return _webElementImplementation.FindElements(@by);
+        }
+
+        public void Clear()
+        {
+            _webElementImplementation.Clear();
+        }
+
+        public void SendKeys(string text)
+        {
+            _webElementImplementation.SendKeys(text);
+        }
+
+        public void Submit()
+        {
+            _webElementImplementation.Submit();
+        }
+
+        public void Click()
+        {
+            try
+            {
+                _webElementImplementation.Click();
+            }
+            catch (Exception)
+            {
+                try
+                {
+                    _actions
+                        .MoveToElement(_webElementImplementation)
+                        .Click()
+                        .Build()
+                        .Perform();
+                }
+                catch (Exception)
```

Comment:
https://stackoverflow.com/questions/2416316/why-is-the-catchexception-almost-always-a-bad-idea